src/csync/csync_exclude.cpp

```diff
@@ -322,10 +309,12 @@ static CSYNC_EXCLUDE_TYPE _csync_excluded_common(c_strlist_t *excludes, const ch
 #endif
 
     /* We create a desktop.ini on Windows for the sidebar icon, make sure we don't sync them. */
-    rc = csync_fnmatch("Desktop.ini", bname, 0);
-    if (rc == 0) {
-        match = CSYNC_FILE_SILENTLY_EXCLUDED;
-        goto out;
+    if (blen == 11) {
+        rc = csync_fnmatch("Desktop.ini", bname, 0);
```

Comment:
why not bname == "Desktop.ini"   (with Qt::CaseInsensitive for file system that needs it) ?

But also, why should we not put this and all other builtin in the regexp?

Reply:
Yes, that's definitely worthy of further simplification. Some of the patterns above have similar opportunities. Adding to the regex doesn't work easily since these want to be SILENTLY_EXCLUDED.